phoenix-core/src/main/java/org/apache/phoenix/coprocessor/IndexRebuildRegionScanner.java

```diff
@@ -613,39 +605,132 @@ private boolean isDeleteFamilyVersion(Mutation mutation) {
         }
         return getMutationsWithSameTS(put, del);
     }
+    /**
+     * In this method, the actual list is repaired in memory using the expected list which is actually the output of
+     * rebuilding the index table row. The result of this repair is used only for verification.
+     */
+    private void repairActualMutationList(List<Mutation> actualMutationList, List<Mutation> expectedMutationList)
+            throws IOException {
+        // Find the first (latest) actual unverified put mutation
+        for (Mutation actual : actualMutationList) {
+            if (actual instanceof Put && !isVerified((Put) actual)) {
+                long ts = getTimestamp(actual);
+                int expectedIndex;
+                int expectedListSize = expectedMutationList.size();
+                for (expectedIndex = 0; expectedIndex < expectedListSize; expectedIndex++) {
+                    if (getTimestamp(expectedMutationList.get(expectedIndex)) <= ts) {
+                        if (expectedIndex > 0) {
+                            expectedIndex--;
+                        }
+                        break;
+                    }
+                }
+                if (expectedIndex == expectedListSize) {
+                    continue;
+                }
+                for (; expectedIndex < expectedListSize; expectedIndex++) {
+                    Mutation mutation = expectedMutationList.get(expectedIndex);
+                    if (mutation instanceof Put) {
+                        mutation = new Put((Put) mutation);
+                    } else {
+                        mutation = new Delete((Delete) mutation);
+                    }
+                    actualMutationList.add(mutation);
```

Comment:
will it be a problem to modify the list when iterating it?
https://stackoverflow.com/questions/6866238/concurrent-modification-exception-adding-to-an-arraylist
https://www.baeldung.com/java-concurrentmodificationexception

Reply:
Yes, this should be a problem. Good catch! I will fix it.

Reply:
This is in general a problem. I was thinking why it did not cause an issue in my tests. The reason is that we are not actually modifying the list here while iterating over it. We modify the list and then terminate the loop. Therefore, it is not a problem. Regardless, I will change the code to eliminate the confusion.